src/widgets/queryinput.coffee

```diff
@@ -50,13 +62,15 @@ class QueryInput extends Widget
   ###
   init: ->
     unless @initialized
-      @element.on "input", (=> @__scheduleUpdate())
-
-      unless (@element.get 0).tagName.toUpperCase() is "TEXTAREA"
-        @element.on "keydown", (e) =>
-          if e.keyCode? and e.keyCode is 13
-            @__scheduleUpdate 0, true
-            @trigger "df:input:submit", [@value]
+      @element.on "input", (event) =>
+        @currentElement = $ event.target
+        # @element.val @value # Synchronize text inputs
+        @__scheduleUpdate()
+
+      @element.filter(":not(textarea)").on "keydown", (event) =>
+        if event.keyCode? and event.keyCode is 13
+          @__scheduleUpdate 0, true
+          @trigger "df:input:submit", [@value]
```

Comment:
I think a test is required for this to ensure that the value sent is the correct (`@currentElement` is the input and not a textarea). It may not be easy.

Reply:
Already have a test for this.